collectors/perf.go

```diff
@@ -319,78 +135,36 @@ func (c *PerfCollector) Collect(ch chan<- prometheus.Metric) {
 // The corresponding metric values are sent separately.
 func (c *PerfCollector) Describe(ch chan<- *prometheus.Desc) {
 
-	ch <- c.AvgDiskBytesPerRead
-	ch <- c.AvgDiskBytesPerRead_Base
-	ch <- c.AvgDiskBytesPerTransfer
-	ch <- c.AvgDiskBytesPerTransfer_Base
-	ch <- c.AvgDiskBytesPerWrite
-	ch <- c.AvgDiskBytesPerWrite_Base
-	ch <- c.AvgDiskQueueLength
-	ch <- c.AvgDiskReadQueueLength
-	ch <- c.AvgDiskSecPerRead
-	ch <- c.AvgDiskSecPerRead_Base
-	ch <- c.AvgDiskSecPerTransfer
-	ch <- c.AvgDiskSecPerTransfer_Base
-	ch <- c.AvgDiskSecPerWrite
-	ch <- c.AvgDiskSecPerWrite_Base
-	ch <- c.AvgDiskWriteQueueLength
-	ch <- c.CurrentDiskQueueLength
-	ch <- c.DiskBytesPerSec
-	ch <- c.DiskReadBytesPerSec
-	ch <- c.DiskReadsPerSec
-	ch <- c.DiskTransfersPerSec
-	ch <- c.DiskWriteBytesPerSec
-	ch <- c.DiskWritesPerSec
-	ch <- c.FreeMegabytes
-	ch <- c.PercentDiskReadTime
-	ch <- c.PercentDiskReadTime_Base
-	ch <- c.PercentDiskTime
-	ch <- c.PercentDiskTime_Base
-	ch <- c.PercentDiskWriteTime
-	ch <- c.PercentDiskWriteTime_Base
-	ch <- c.PercentFreeSpace
-	ch <- c.PercentFreeSpace_Base
-	ch <- c.PercentIdleTime
-	ch <- c.PercentIdleTime_Base
+	ch <- c.QueueLength
+	ch <- c.ReadBytesPerSec
+	ch <- c.ReadsPerSec
+	ch <- c.WriteBytesPerSec
+	ch <- c.WritesPerSec
+	ch <- c.FreeBytes
+	ch <- c.ReadTime
+	ch <- c.WriteTime
+	ch <- c.FreeSpace
+	ch <- c.IdleTime
 	ch <- c.SplitIOPerSec
 }
 
 type Win32_PerfRawData_PerfDisk_LogicalDisk struct {
-	Name                         string
-	AvgDiskBytesPerRead          uint64
-	AvgDiskBytesPerRead_Base     uint32
-	AvgDiskBytesPerTransfer      uint64
-	AvgDiskBytesPerTransfer_Base uint32
-	AvgDiskBytesPerWrite         uint64
-	AvgDiskBytesPerWrite_Base    uint32
-	AvgDiskQueueLength           uint64
```

Comment:
It turns out that this is a Counter - which makes no sense for a queue length,

What it actually is is disk requests queued.